### ISSUE TITLE
Introduce _ugp.py

### ISSUE THREAD

**Issue by devinrsmith** [deephaven/deephaven-core#2364]
We should simplify our usage of UGP from python; for internal purposes initially, but eventually for external users.

```python
with _ugp.exclusive_lock():
  ...
```

```python
with _ugp.shared_lock():
  ...
```

The general code for this has been sketched out in testbase.py using `@contextlib.contextmanager`, but note, the lock() should be *outside* of the try instead of inside.

**Comment by devinrsmith** [comment]
Relevant for #2321 2321

**Comment by jmao-denver** [comment]
resolved by #2321
